pkg/chaincode/example_test.go

```diff
@@ -29,6 +31,7 @@ func Example() {
 	// gRPC connection a target peer.
 	connection := newGrpcConnection()
 	defer connection.Close()
+	Endorser := peer.NewEndorserClient(connection)
```

Comment:
Absolutely do **not** do this! You are giving the caller a lot more work, a lot more dependencies, and a lot more concepts to deal with than necessary. All they need to provide is the gRPC connection. The caller should not be made to deal with gRPC client stubs for specific Fabric services.